dmipy/core/modeling_framework.py

```diff
@@ -2271,16 +2282,83 @@ def amico_idx(self):
+                    multi-compartment model
+            Returns:
+                observation matrix M
+        """
+
+        dir_params = [p for p in self.mc_model.parameter_names
+                      if p.endswith('mu')]
+        if len(dir_params) != len(model_dirs):
+            raise ValueError("Length of model_dirs should correspond "
+                             "to the number of directional parameters!")
+
+        if not self._freezed_parameters_vector:
+            self._amico_grid, self._amico_idx = {}, {}
+
+            grid_params = \
+                [p for p in self.mc_model.parameter_names
+                 if not p.endswith('mu') and
+                 not p.startswith('partial_volume')]
+
+            # Compute length of the vector x0
+            x0_len = 0
+            for m_idx in range(self.N_models):
+                m_atoms = 1
+                for p in self.models[m_idx].parameter_names:
+                    if self.mc_model.model_names[m_idx] + p in grid_params:
+                        m_atoms *= self.Nt
+                x0_len += m_atoms
+
+            for m_idx in range(self.N_models):
+                model = self.models[m_idx]
+                model_name = self.mc_model.model_names[m_idx]
+
+                param_sampling, grid_params_names = [], []
+                m_atoms = 1
+                for p in model.parameter_names:
+                    if model_name + p not in grid_params:
+                        continue
+                    grid_params_names.append(model_name + p)
+                    p_range = self.mc_model.parameter_ranges[model_name + p]
+                    self._amico_grid[model_name + p] = \
+                        np.full(x0_len, np.mean(p_range))
+                    param_sampling.append(np.linspace(p_range[0], p_range[1],
+                                                      self.Nt, endpoint=True))
+                    m_atoms *= self.Nt
+
+                self._amico_idx[model_name] =\
+                    sum([len(self._amico_idx[k])
+                         for k in self._amico_idx]) + \
+                    np.arange(m_atoms)
+                params_mesh = np.meshgrid(*param_sampling)
+                for p_idx, p in enumerate(grid_params_names):
+                    self.grids[p][self._amico_idx[model_name]] =\
+                        np.ravel(params_mesh[p_idx])
+
+                self._amico_grid['partial_volume_' + str(m_idx)] = \
+                    np.zeros(x0_len)
+                self._amico_grid['partial_volume_' +
+                                 str(m_idx)][self._amico_idx[model_name]] = 1.
+            self._freezed_parameters_vector = True
+
+        for d_idx, dp in enumerate(dir_params):
+            self._amico_grids[dp] = model_dirs[d_idx]
+
+        # QUESTION:
+        # Should we return simulated signals with b=0 values
+        # or only diffusion weighted, I think b=0 impacts a lot
+        # results when solving nnls
```

Comment:
why should it matter to nnls?

Reply:
Because b=0 have higher values than diffusion weighted values (and less affected by noise), so it will force that sum of b=0 values of atoms is equal to 1 and in this way will prevent that some atoms are fitted to noise. Also, there is a difference if we use for example 18 b=0 values (as in HCP dataset) or just single b=0 value that is equal to the mean of all b=0 values (that would be 1 after normalization), because the higher the number of b=0 values, nnls will give more importance to fitting b=0 parts of the atoms.

Reply:
So you vote for keeping the b=0 measurements then, if I understand correctly?

Reply:
I agree with this also.

Reply:
Yes, but maybe it is something that we can investigate

Reply:
I think we can leave the default as being just sample anything that is in the acquisition scheme.
we can test the effect of having more B0s by just making a scheme with more or less b0s in it.